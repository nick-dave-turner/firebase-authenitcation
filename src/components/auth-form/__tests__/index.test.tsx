import React from "react";
import { MemoryRouter } from "react-router-dom";
import { act, fireEvent, render } from "@testing-library/react";
import pretty from "pretty";

import { AuthForm } from "..";

const inputs = { email: "", password: "" };
const handleOnChange = jest.fn();
const handleOnSubmit = jest.fn();

describe("AuthForm component", () => {
  it("should render when type === SignIn", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/mock-route"]}>
        <AuthForm
          errorMessage=""
          type="SIGNIN"
          handleOnChange={handleOnChange}
          handleOnSubmit={handleOnSubmit}
          inputs={inputs}
          isFormValid={false}
        />
      </MemoryRouter>
    );
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it("should render when type === SignUp", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/mock-route"]}>
        <AuthForm
          errorMessage=""
          type="SIGNUP"
          handleOnChange={handleOnChange}
          handleOnSubmit={handleOnSubmit}
          inputs={inputs}
          isFormValid={false}
        />
      </MemoryRouter>
    );
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it("should render when error message present", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/mock-route"]}>
        <AuthForm
          errorMessage="Mock error message"
          type="SIGNIN"
          handleOnChange={handleOnChange}
          handleOnSubmit={handleOnSubmit}
          inputs={inputs}
          isFormValid={false}
        />
      </MemoryRouter>
    );
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it("should submit form when valid", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/mock-route"]}>
        <AuthForm
          errorMessage=""
          type="SIGNIN"
          handleOnChange={handleOnChange}
          handleOnSubmit={handleOnSubmit}
          inputs={inputs}
          isFormValid={false}
        />
      </MemoryRouter>
    );

    const submitButton = getByTestId("submit-button");
    const emailCntentInput = getByTestId("email-input");
    const passwordContentInput = getByTestId("password-input");

    act(() => {
      fireEvent.change(emailCntentInput, { target: { value: "a" } });
      fireEvent.change(passwordContentInput, { target: { value: "b" } });
      fireEvent.click(submitButton);
    });

    expect(handleOnChange).toHaveBeenCalledTimes(2);
    expect(handleOnSubmit).toHaveBeenCalled();
  });
});
