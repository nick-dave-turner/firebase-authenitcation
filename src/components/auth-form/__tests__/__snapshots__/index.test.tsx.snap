// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm component should render when error message present 1`] = `
"<main class=\\"MuiContainer-root makeStyles-layout-11 MuiContainer-maxWidthSm\\">
  <div class=\\"MuiPaper-root makeStyles-paper-12 MuiPaper-elevation3 MuiPaper-rounded\\">
    <h1 class=\\"MuiTypography-root MuiTypography-h5\\">Welcome back to FireBase Demo</h1>
    <h2 class=\\"MuiTypography-root MuiTypography-subtitle1\\">Sign into your account to continue</h2>
    <form class=\\"makeStyles-form-13\\" novalidate=\\"\\">
      <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"email\\" id=\\"email-label\\">Email Address<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"email\\" id=\\"email\\" name=\\"email\\" required=\\"\\" type=\\"text\\" data-testid=\\"email-input\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div>
      </div>
      <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"password\\" id=\\"password-label\\">Password<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"current-password\\" id=\\"password\\" name=\\"password\\" required=\\"\\" type=\\"password\\" data-testid=\\"password-input\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div>
      </div>
      <div class=\\"MuiPaper-root MuiAlert-root MuiAlert-standardError makeStyles-alert-15 MuiPaper-elevation0\\" role=\\"alert\\">
        <div class=\\"MuiAlert-icon\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
            <path d=\\"M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"></path>
          </svg></div>
        <div class=\\"MuiAlert-message\\">Mock error message</div>
      </div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-14 MuiButton-containedPrimary MuiButton-fullWidth\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"submit-button\\"><span class=\\"MuiButton-label\\">Sign In</span><span class=\\"MuiTouchRipple-root\\"></span></button>
    </form>
    <div class=\\"MuiGrid-root MuiGrid-container\\">
      <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"></div>
      <div class=\\"MuiGrid-root MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary\\" href=\\"/signup\\">Don't have an account? Sign Up</a></div>
    </div>
  </div>
</main>"
`;

exports[`AuthForm component should render when type === SignIn 1`] = `
"<main class=\\"MuiContainer-root makeStyles-layout-1 MuiContainer-maxWidthSm\\">
  <div class=\\"MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded\\">
    <h1 class=\\"MuiTypography-root MuiTypography-h5\\">Welcome back to FireBase Demo</h1>
    <h2 class=\\"MuiTypography-root MuiTypography-subtitle1\\">Sign into your account to continue</h2>
    <form class=\\"makeStyles-form-3\\" novalidate=\\"\\">
      <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"email\\" id=\\"email-label\\">Email Address<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"email\\" id=\\"email\\" name=\\"email\\" required=\\"\\" type=\\"text\\" data-testid=\\"email-input\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div>
      </div>
      <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"password\\" id=\\"password-label\\">Password<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"current-password\\" id=\\"password\\" name=\\"password\\" required=\\"\\" type=\\"password\\" data-testid=\\"password-input\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div>
      </div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"submit-button\\"><span class=\\"MuiButton-label\\">Sign In</span><span class=\\"MuiTouchRipple-root\\"></span></button>
    </form>
    <div class=\\"MuiGrid-root MuiGrid-container\\">
      <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"></div>
      <div class=\\"MuiGrid-root MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary\\" href=\\"/signup\\">Don't have an account? Sign Up</a></div>
    </div>
  </div>
</main>"
`;

exports[`AuthForm component should render when type === SignUp 1`] = `
"<main class=\\"MuiContainer-root makeStyles-layout-6 MuiContainer-maxWidthSm\\">
  <div class=\\"MuiPaper-root makeStyles-paper-7 MuiPaper-elevation3 MuiPaper-rounded\\">
    <h1 class=\\"MuiTypography-root MuiTypography-h5\\">Welcome to Firebase Demo</h1>
    <h2 class=\\"MuiTypography-root MuiTypography-subtitle1\\">Create an account to start using Firebase Demo</h2>
    <form class=\\"makeStyles-form-8\\" novalidate=\\"\\">
      <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"email\\" id=\\"email-label\\">Email Address<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"email\\" id=\\"email\\" name=\\"email\\" required=\\"\\" type=\\"text\\" data-testid=\\"email-input\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div>
      </div>
      <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"password\\" id=\\"password-label\\">Password<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"current-password\\" id=\\"password\\" name=\\"password\\" required=\\"\\" type=\\"password\\" data-testid=\\"password-input\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div>
      </div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-9 MuiButton-containedPrimary MuiButton-fullWidth\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"submit-button\\"><span class=\\"MuiButton-label\\">Sign Up</span><span class=\\"MuiTouchRipple-root\\"></span></button>
    </form>
    <div class=\\"MuiGrid-root MuiGrid-container\\">
      <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"></div>
      <div class=\\"MuiGrid-root MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary\\" href=\\"/signin\\">Already have an account? Sign In</a></div>
    </div>
  </div>
</main>"
`;
